src/Framework/AspNetCoreAnalyzers/src/CodeFixes/Dependencies/ThisAndExtensionMethod.cs

```diff
@@ -0,0 +1,25 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using Microsoft.CodeAnalysis;
+
+namespace Microsoft.AspNetCore.Analyzers;
+
+internal readonly struct ThisAndExtensionMethod(ITypeSymbol thisType, string extensionMethod)
+{
+    public ITypeSymbol ThisType { get; } = thisType;
+    public string ExtensionMethod { get; } = extensionMethod;
+
+    public override bool Equals(object? obj)
+    {
+        if (obj is ThisAndExtensionMethod other)
+        {
+            return SymbolEqualityComparer.Default.Equals(ThisType, other.ThisType) &&
+                ExtensionMethod.Equals(other.ExtensionMethod, StringComparison.OrdinalIgnoreCase);
+        }
+        return false;
+    }
+
+    public override int GetHashCode() => HashCode.Combine(ThisType, ExtensionMethod);
```

Comment:
Should it hash `ExtensionMethod.ToLowerInvariant()` instead since the equality comparison is done with `OrdinalIgnoreCase`?

And I assume that's why it's not a `record struct`?

Reply:
Yeah, although I suppose a strict ordinal comparison is the right way to go here given `mapopenapi` and `MapOpenApi` aren't equivalent invocations.

Reply:
Now I believe just having a `record struct` might even be better, since:
1- you won't have to write anything, just `internal readonly record struct ThisAndExtensionMethod(ITypeSymbol thisType, string extensionMethod);`
2- it will implement `IEquatable<ThisAndExtensionMethod>` which won't box, and will be used by the dictionary, unlike `Equals(object)` which will box.